gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/api/ApiMongoRepositoryImpl.java

```diff
@@ -101,6 +101,27 @@ public Page<String> searchIds(List<ApiCriteria> apiCriteria, Pageable pageable,
         return new Page<>(apisIds, pageable.pageNumber(), apisIds.size(), total);
     }
 
+    @Override
+    public List<ApiMongo> findAllNames(ApiCriteria apiCriteria, Sortable sortable) {
+        final Query query = new Query();
+        query.fields().include("_id", "name");
+
+        if (apiCriteria != null) {
+            fillQuery(query, List.of(apiCriteria));
+        }
+
+        Sort sort;
+        if (sortable == null) {
+            sort = Sort.by(ASC, "name");
+        } else {
+            Sort.Direction sortOrder = sortable.order().equals(Order.ASC) ? ASC : Sort.Direction.DESC;
```

Comment:
Maybe `Order.ASC.equals(sortable.order())`to prevent null pointer

Reply:
Done